coderd/provisionerjobs.go

```diff
@@ -343,3 +338,46 @@ func convertProvisionerJob(provisionerJob database.ProvisionerJob) codersdk.Prov
 func provisionerJobLogsChannel(jobID uuid.UUID) string {
 	return fmt.Sprintf("provisioner-log-logs:%s", jobID)
 }
+
+// provisionerJobLogsMessage is the message type published on the provisionerJobLogsChannel() channel
+type provisionerJobLogsMessage struct {
+	EndOfLogs bool                         `json:"end_of_logs,omitempty"`
+	Logs      []database.ProvisionerJobLog `json:"logs,omitempty"`
+}
+
+func (api *API) followLogs(jobID uuid.UUID) (chan database.ProvisionerJobLog, func(), error) {
```

Comment:
Might as well limit what the caller can do with the channel
```suggestion
func (api *API) followLogs(jobID uuid.UUID) (<-chan database.ProvisionerJobLog, func(), error) {
```